### PR TITLE
Add tox.ini for flake8 control

### PR DESCRIPTION
**Pull request by jwiggins**
We're not ready to run flake8 on the code regularly yet, but with #557 and #558 in place we can now work towards that goal. To that end, I'm introducing a `tox.ini` config which adds ignores for machine generated and vendored files that we don't care to see linter errors for.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[flake8]
+application-package-names = enable,kiva,pyface,traits,traitsui
```
Comment (rahulporuri) [inline tox.ini:2]: do we want to add the `application-import-names` and put `enable` and `kiva` under there instead? -> This is bound to raise a lot of new flake8 errors though.

```suggestion
application-package-names = pyface,traits,traitsui
application-import-names = enable,kiva
```

**Review by rahulporuri — APPROVED** [review]
LGTM

File: tox.ini (hunk 1 above)
Comment (jwiggins) [reply tox.ini:2]: Good catch

**Review by rahulporuri — APPROVED** [review]
Still LGTM

**Comment by jwiggins** [comment]
Thanks for the feedback

**Pull request merged**